clearwater-infrastructure/usr/share/clearwater/bin/cw-flag

```diff
@@ -0,0 +1,81 @@
+#!/bin/bash
+
+# @file cw-flag
+#
+# Copyright (C) Metaswitch Networks 2017
+# If license terms are provided to you in a COPYING file in the root directory
+# of the source code repository by which you are accessing this code, then
+# the license outlined in that COPYING file applies to your use.
+# Otherwise no rights are granted except for those provided to you by
+# Metaswitch Networks in a separate written agreement.
+
+# Utility script which allows semi-permanent flags to be set and cleared.
+
+FLAG_DIR=/tmp/cw-flags
+USAGE="Usage: flag {clear|set|read} <flag_name>, or flag timeout <flag_name> <timeout (s)>" >&2
+DEFAULT_TIMEOUT=60
+
+method=$1
+flag_file=$FLAG_DIR/$2
+
+if [ -z "$3" ]; then
+  timeout=$DEFAULT_TIMEOUT
+else
+  timeout=$3
+fi
+
+# Clears the flag by removing the flag file.
+clear_flag() {
+  rm -f $flag_file
+}
+
+# Clears the flag if it has been set for longer than $timeout.
+timeout_flag() {
+  if [ -f $flag_file ]; then
+    flag_time=$( expr `date +%s` - `stat -c %Y $flag_file` )
+    if [ $flag_time -gt $timeout ]; then
+      clear_flag
+    fi
+  fi
+}
+
+# Sets the flag by (re)creating the flag file.
+set_flag() {
+  clear_flag  # First clear the flag
+  mkdir -p $FLAG_DIR
+  touch $flag_file
+}
+
+# Reads the flag, returning 1 if the flag is set and 0 if it is clear.
+read_flag() {
+  if [ -f $flag_file ]; then
```

Comment:
Can this be simplified - something more like `return -f $flag_file`

Reply:
It doesn't seem like there's a clear accepted way to do this (`return -f $flag_file` and variations give a `numeric argument required` error), so I'll leave this as-is. It's a bit of extra code, but I think it's clear exactly what value it's returning.